framework/Optimizers/GradientBasedOptimizer.py

```diff
@@ -115,6 +134,19 @@ def localInputAndChecks(self, xmlNode):
         self.gainShrinkFactor = float(gainShrinkFactor.text) if gainShrinkFactor is not None else 2.0
       except ValueError:
         self.raiseAnError(ValueError, 'Not able to convert <gainShrinkFactor> into a float.')
+
+      centralDiff = convergence.find('centralDifference')
+      try:
+        self.useCentralDiff = (centralDiff.text.strip().lower() in utils.stringsThatMeanTrue()) if centralDiff is not None else False
+      except ValueError:
+        self.raiseAnError(ValueError, 'Not able to convert <centralDifference> into a boolean.')
+
+      useGradHist = convergence.find('useGradientHistory')
+      try:
+        self.useGradHist = (useGradHist.text.strip().lower() in utils.stringsThatMeanTrue()) if centralDiff is not None else False
+      except ValueError:
+        self.raiseAnError(ValueError, 'Not able to convert <useGradientHistory> into a boolean.')
```

Comment:
see above. Why not input parameter?

Reply:
So, it looks like the GradientBasedOptimizer and its inheritors have never used the paramInputs, so this is just continuing with what was already there.

That said, it does need fixing.

By way of note, fixing this includes changing:
 - `Optimizer._readMoreXML`, the `_readMoreXMLbase` should return the `paramInput`
 - Instead of sending `xmlNode` to `localInputAndChecks`, only the `paramInput` should be sent
 - All of the optimizers (GradientBased, FiniteDifference, SPSA, ConjugateGradient) need to have their `localInputAndChecks` rewritten to use the input specs.

This does need to happen, this might be a large request before accepting this PR.

Reply:
If it is possible, I would like to see this change in this PR. It should not be too bad I hope.